packaging/macos/utility.m

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+
+#import <Cocoa/Cocoa.h>
+#include <dlfcn.h>
+#include <libgen.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <sys/resource.h>
+#include <mach/machine.h>
+
+#define SYSTEM_MONO_PATH @"/Library/Frameworks/Mono.framework/Versions/Current/"
+#define SYSTEM_MONO_MIN_VERSION @"6.4"
+#define DOTNET_MIN_MACOS_VERSION 10.15
+
+typedef void* hostfxr_handle;
+struct hostfxr_initialize_parameters
+{
+	size_t size;
+	char *host_path;
+	char *dotnet_root;
+};
+
+typedef int32_t(*hostfxr_initialize_for_dotnet_command_line_fn)(
+	int argc,
+	char **argv,
+	struct hostfxr_initialize_parameters *parameters,
+	hostfxr_handle *host_context_handle);
+
+typedef int32_t(*hostfxr_run_app_fn)(const hostfxr_handle host_context_handle);
+typedef int32_t(*hostfxr_close_fn)(const hostfxr_handle host_context_handle);
+typedef int (* mono_main)(int argc, char **argv);
+
+int main(int argc, char **argv)
+{
+	BOOL useMono = NO;
+
+	if (@available(macOS 10.15, *))
+		useMono = [[[NSProcessInfo processInfo] environment]objectForKey:@"OPENRA_PREFER_MONO"] != nil;
+	else
+		useMono = YES;
+
+	if (useMono)
```

Comment:
Mono is only compiled for intel, so this probably needs a not-arm check.

Reply:
Should I add that check to the Launcher as well?

Reply:
The launcher handles this by compiling the mono launcher as intel only, allowing/forcing it to run through rosetta.

Reply:
made it run `useMono && !arm`